### PR TITLE
Poco Path and simbol $

### PR DESCRIPTION
**Pull request by bas524**
bug : into current implementation od path_unix.cpp I can't use simbol $ like a simbol, not a variable prefix, for example,
I need to use path /home/user/O1$$/folder. I propose to use escape sequence for this, for ex., if I want to use Poco::Glob, I can do 
```
std::set<std::string> files;
std::string path = "/home/user/O1$$/folder"
Poco::replaceInPlace(path, "$", "\\$");
Poco::Glob::glob(path, files);
```

### CODE REVIEW

**Comment by bas524** [comment]
Hello, please look at the AppVeyor build, it contain errors with odbc test building, not with my commit

**Comment by zosrothko** [comment]
The CI AppVeyor & Travis are in a unstable state because there is a switch to the c++11 standard for the develop branch on the way. This should be fixed sooner or later.

File: Data/SQLite/src/SessionImpl.cpp [hunk 1]
```diff
@@ -207,12 +207,14 @@ void SessionImpl::close()
 {
 	if (_pDB) {
 		int result = 0;
+		int times = 50;
 		do {
```
Comment (aleks-f) [inline Data/SQLite/src/SessionImpl.cpp]: please check [Coding Style Guide](https://www.appinf.com/download/CppCodingStyleGuide.pdf)

File: Foundation/src/Path_UNIX.cpp [hunk 2]
```diff
@@ -120,7 +120,15 @@ std::string PathImpl::expandImpl(const std::string& path)
...
 	{
-		if (*it == '$')
+		if (*it == '\\')
+		{
+			++it;
+			if (*it == '$')
+			{
+				result += *it++;
+			}
+		}
```
Comment (aleks-f) [inline Foundation/src/Path_UNIX.cpp]: please add a test case for this feature

**Comment by bas524** [comment]
Ok, I fixed codestyle and add testcase for Path::expand

File: Foundation/src/Path_UNIX.cpp (hunk 2 above)
Comment (bas524) [inline Foundation/src/Path_UNIX.cpp]: PLease, review my changes

File: Data/SQLite/src/SessionImpl.cpp [hunk 3]
```diff
@@ -205,16 +205,20 @@ void SessionImpl::open(const std::string& connect)
...
+	if (_pDB) 
+	{
 		int result = 0;
 		int times = 50;
-		do {
+		do 
+		{
 			result = sqlite3_close(_pDB);
-			if (result == SQLITE_BUSY) {
+			if (result == SQLITE_BUSY)
```
Comment (aleks-f) [inline Data/SQLite/src/SessionImpl.cpp]: Looping limited number of times prevents endless loop but does not necessarily fix the root cause for the busy state. If this is happening, something is [still opened](http://www.sqlite.org/c3ref/close.html) - likely a prepared statement, which should be finalized (or should not be lingering to start with).

Comment (bas524) [inline Data/SQLite/src/SessionImpl.cpp]: So, if I want to close session correctly, then I must to have list of all statements, and into close method all statements must be destroyed before sqlite3_close. And every statement  into destuctor must erase itself from session which owned statements-list. Am I right?

Comment (aleks-f) [inline Data/SQLite/src/SessionImpl.cpp]: you can loop through all prepared statements using
```
sqlite3_next_stmt(_pDb, 0);
```
and then finalize each.
Note that there may be other `SQLITE_BUSY` on close reasons, like opened blob handles or backups. a proper solution would be to simulate such conditions and make sure they are handled correctly; preferably, in a separate pull.

**Comment by bas524** [comment]
As for sessionimpl->close... I add finalize for every busy-stmt, but I can't get opened blobs and backups now ...

**Comment by aleks-f** [comment]
includes fix for #1373

**Comment by zosrothko** [comment]
@bas524 The test below is failing on Windows. can youy fix it?
```
void PathTest::testExpandUnix()
{
	std::string pathWithoutVar = "/usr/share/O1\\$\\$/folder";
	std::string pathWithVar = "${HOME}/folder";
	Path p;
	std::string s = p.expand(pathWithoutVar);
	assert (s == "/usr/share/O1$$/folder");
	s = p.expand(pathWithVar);
	Path tmpPath = Path::home();
	tmpPath.append("folder");
	assert (s == tmpPath.toString());
}
```
```
C:\Users\fandre\Documents\git\poco-develop>build\script\runtests2.cmd Foundation testExpandUnix


****************************************
*** Foundation
****************************************
testExpandUnix: ===================
p=/usr/share/O1$$/folder
s=/usr/share/O1\$\$/folder
FAILURE
!!!FAILURES!!!
Runs: 1   Failures: 1   Errors: 0

There was 1 failure:
 1: class CppUnit::TestCaller<class PathTest>.testExpandUnix
    "s == "/usr/share/O1$$/folder""
    in "src\PathTest.cpp", line 388
1 runs, 1 failed.
```

**Comment by bas524** [comment]
Yes, shure.

**Comment by bas524** [comment]
Complete. Request #1850, please review it

**Pull request merged**